integration-libs/opf/base/root/services/opf-payment-metadata-store.service.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { OpfPaymentMetadata, PaymentProcessingState } from '../model/opf.model';
+
+const initialState = {
+  termsAndConditionsChecked: false,
+  selectedPaymentOptionId: undefined,
+  paymentProcessingState: PaymentProcessingState.NOT_IN_PROGRESS,
+};
+
+@Injectable({ providedIn: 'root' })
+export class OpfPaymentMetadataStoreService {
+  opfPaymentMetadataState = new BehaviorSubject<OpfPaymentMetadata>(
+    initialState
+  );
+
+  getOpfMetadataState(): Observable<OpfPaymentMetadata> {
+    return this.opfPaymentMetadataState.asObservable();
+  }
+
+  opfMetadataUpdate(payload: Partial<OpfPaymentMetadata>): void {
```

Comment:
I think the name "updateOpfMetadata" here would be more consistent with other methods.